### ISSUE TITLE
No date can be entered

### ISSUE THREAD

**Issue by klaus1902** [meeting-room-booking-system/mrbs-code#3675]
MRBS 1.11.5
php 8.1.28
mysql 8.0.37
Apache Server

I have installed mrbs 1.11.5 on my web server. I have been working with mrbs for some time. Everything works very well. After updating to the latest version, I can no longer enter appointments. The following error is displayed:

Uncaught exception ‘TypeError’ in /var/home/xxxxxx.at/www/kegel/mrbs/functions.inc at line 928
abs(): Argument #1 ($num) must be of type int|float, string given
#0 /var/home/xxxxxx.at/www/kegel/mrbs/functions.inc(928): abs(‘7200’)
#1 /var/home/xxxxxx.at/www/kegel/mrbs/edit_entry.php(1633): MRBS\toTimeString(‘7200’, NULL)
#2 {main}

MRBS GET: Array
(
    [view] => day
    [year] => 2024
    [month] => 5
    [day] => 22
    [area] => 1
    [room] => 1
    [hour] => 12
    [minute] => 0
)
MRBS POST: Array
(
)
MRBS SESSION: Array
(
    [last_page] => /kegel/mrbs/index.php?MAX_FILE_SIZE=268435456&csrf_token=32a666006d7098c9e26fd7706cbff9c13f6c41156a022cd3b54dc48a3e0336f7&view=day&area=1&page_date=2024-05-22
    [this_page] => /kegel/mrbs/edit_entry.php?view=day&year=2024&month=5&day=22&area=1&room=1&hour=12&minute=0
    [csrf_token] => 32a666006d7098c9e26fd7706cbff9c13f6c41156a022cd3b54dc48a3e0336f7
    [user] => MRBS\User Object
        (
            [username] => xxxxx
            [display_name] => xxxxx
            [email] => xxxxx
            [level] => 1
            [data:protected] => Array
                (
                )

        )

)

Unfortunately, I no longer know what to do.
Thank you for your help

**Comment by jberanek** [comment]
So, it would seem like somehow your database schema is storing attributes of areas as strings, rather than as integers, or somehow they're at least being returned as strings rather than integers. Can you check these fields in your 'mrbs_area' table?

![chrome_NnlzFOwZr3](https://github.com/meeting-room-booking-system/mrbs-code/assets/571708/24fb269a-c674-40b7-91eb-26d93badbd42)

**Comment by campbell-m** [comment]
No, it's PHP PDO MySQL which returns ints as strings.  Fix coming up soon.

**Comment by jberanek** [comment]
> No, it's PHP PDO MySQL which returns ints as strings. Fix coming up soon.

Really? It's working for me, guessing because of my ancient PHP? (I really must fix that!)

```
Database	MariaDB package 10.6.15-MariaDB
System	Linux linda 6.3.3-lp154.5-default #1 SMP PREEMPT_DYNAMIC TKG Wed May 17 19:16:13 UTC 2023 (1237d35 x86_64
Server time	Wednesday, May 22, 2024 at 8:22:26 PM GMT+1
Server software	Apache/2.4.51 (Linux/SUSE)
PHP	7.4.33
```

**Comment by jberanek** [comment]
Still working for me after an upgrade to PHP 8.1 too, so also down to the MySQL/Maria driver?

Database	MariaDB package 10.6.15-MariaDB
System	Linux linda 6.3.3-lp154.5-default #1 SMP PREEMPT_DYNAMIC TKG Wed May 17 19:16:13 UTC 2023 (1237d35 x86_64
Server time	Wednesday, May 22, 2024 at 8:37:27 PM GMT+1
Server software	Apache/2.4.51 (Linux/SUSE)
PHP	8.1.28
Extensions	Core, PDO, Phar, Reflection, SPL, SimpleXML, bz2, cgi-fcgi, ctype, curl, date, dom, fileinfo, filter, gd, gettext, gmp, hash, iconv, intl, json, ldap, libxml, mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_pgsql, pdo_sqlite, pgsql, posix, session, snmp, sockets, sqlite3, standard, tokenizer, xml, xmlreader, xmlwriter, zip, zlib```

```

**Comment by campbell-m** [comment]
Yes, it depends on the driver.

**Comment by campbell-m** [comment]
Now fixed in [93b417a](https://github.com/meeting-room-booking-system/mrbs-code/commit/93b417a4ce5f8c207d5dd2cc62e16d0fb7742beb).  I recommend you upgrade to the latest code in the main branch by downloading the latest code from [this page](https://github.com/meeting-room-booking-system/mrbs-code) and following the green Code button.

**Comment by klaus1902** [comment]
Thanks for the quick help. It works
